onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.h

```diff
@@ -107,6 +107,7 @@ struct TensorrtFuncState {
   bool engine_decryption_enable;
   int (*engine_decryption)(const char*, char*, size_t*);
   int (*engine_encryption)(const char*, char*, size_t);
+  bool save_engine_to_file; // Whether to save engine to file. Default is false
```

Comment:
do we need a separate bool for this?
are there cases when "engine_cache_enable" is set to true but we wouldn't serialize the engines?

Reply:
we might want to rename the variable though.
it's not just serializing engines. there's also profile.
and in future it's possible there may be other metadata we want to write out?

Reply:
I renamed the variable.
profile cache being updated or not is highly dependent on engine cache. 
I made a comment to that variable for clarity.

Reply:
any thoughts on where the timing cache serialization would happen when implemented?

Reply:
My thought is that timing cache serialization is done in release_state_func() same as engine cache serialization.
And we also need another variable update_timing_cache for timing cache serialization/save. 

timing cache serialization/save will happen in following situations:

1. timing cache is enabled, and there is no timing cache present.
2. timing cache is enabled, and timing cache is present, but engine is rebuilt.